source/common/common/perf_annotation.h

```diff
@@ -86,6 +111,30 @@ class PerfAnnotationContext {
   void record(std::chrono::nanoseconds duration, absl::string_view category,
               absl::string_view description);
 
+  /**
+   * Starts duration measurement for a category and description, which are shown
+   * as separate columns in the generated output table. The measurement is
+   * stopped and recorded with a corresponding end() call. The measurement can
+   * be started and stopped in different scopes, but the flow of these start and
+   * stop points for the same category and description must be strictly sequential.
+   *
+   * @param category the name of a category for the recording.
+   * @param description the name of description for the recording.
+   */
+  void begin(absl::string_view category, absl::string_view description);
+
+  /**
+   * Ends and records duration measurement for a category and description, which are
+   * shown as separate columns in the generated output table. Before ending the
+   * measurement must be started first with a begin() call. The measurement can
+   * be started and ended in different scopes, but the flow of these start and
+   * stop points for the same category and description must be strictly sequential.
```

Comment:
seems like you don't need to repeat all of this paragraph from above, but could just reference the doc in begin().